yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -359,7 +360,9 @@ private[spark] class ApplicationMaster(
               }
             logDebug(s"Number of pending allocations is $numPendingAllocate. " +
                      s"Sleeping for $sleepInterval.")
-            Thread.sleep(sleepInterval)
+            allocatorLock.synchronized {
+              allocatorLock.wait(sleepInterval)
```

Comment:
In this case it might be ok, but a `wait` call should not be called from outside of a `while`: http://stackoverflow.com/questions/1038007/why-should-wait-always-be-called-inside-a-loop